prog/mutation.go

```diff
@@ -337,6 +340,31 @@ func (t *BufferType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []
 	return
 }
 
+func mutateBufferSize(r *randGen, arg *DataArg, minLen, maxLen uint64) {
+	oldSize := arg.Size()
+	for oldSize == arg.Size() {
+		n := uint64(r.Intn(16)) + 1
+		if r.bin() {
+			// increase the size
+			if arg.Size() >= maxLen {
+				continue
+			}
+			if arg.Size()+n > maxLen {
+				n = maxLen
+			}
+		} else {
+			// decrease the size
+			if arg.Size() == 0 {
+				continue
+			}
+			if arg.Size()-n < minLen {
+				n = minLen
+			}
+		}
+		arg.size = n
+	}
+}
```

Comment:
Since we have the loop with oldSize check anyway, this can be simplified to:
```
func mutateBufferSize(r *randGen, arg *DataArg, minLen, maxLen uint64) {	
	for oldSize := arg.Size(); oldSize == arg.Size(); {
		arg.size += uint64(r.Intn(33)) - 16
		if arg.size < minLen {
			arg.size = minLen
		}
		if arg.size > maxLen {
			arg.size = maxLen
		}
	}
}
```